src/Session/SessionManager.cc

```diff
@@ -51,9 +58,9 @@ void SessionManager::OnUpgrade(
         return;
     }
 
-    _session_number++;
-    // protect against overflow
-    _session_number = max(_session_number, 1u);
+    struct timeval tv;
```

Comment:
we already use `uuid.h` for generating unique tokens. We shouldn't reinvent the wheel here

Reply:
That would mean using 128 bit numbers for these ids. I guess we can do that, but it will need more testing.

Reply:
the alternative would be to use `https://www.cplusplus.com/reference/chrono/high_resolution_clock/now/` I guess.